viz_scripts/scaffolding.py

```diff
@@ -161,6 +161,40 @@ def load_viz_notebook_data(year, month, program, study_type, dic_re, dic_pur=Non
 
     return expanded_ct, file_suffix, quality_text, debug_df
 
+def load_viz_notebook_sensor_inference_data(year, month, program, include_test_users=False, sensed_algo_prefix="cleaned"):
+    """ Inputs:
+    year/month/program = parameters from the visualization notebook
+
+    Pipeline to load and process the data before use in sensor-based visualization notebooks.
+    """
+    tq = get_time_query(year, month)
+    participant_ct_df = load_all_participant_trips(program, tq, include_test_users)
+    expanded_ct = participant_ct_df
+    expanded_ct["primary_mode_non_other"] = participant_ct_df.cleaned_section_summary.apply(lambda md: max(md["distance"], key=md["distance"].get))
+    expanded_ct.primary_mode_non_other.replace({"ON_FOOT": "WALKING"}, inplace=True)
+    valid_sensed_modes = ["WALKING", "BICYCLING", "IN_VEHICLE", "AIR_OR_HSR", "UNKNOWN"]
+    expanded_ct["primary_mode"] = expanded_ct.primary_mode_non_other.apply(lambda pm: "OTHER" if pm not in valid_sensed_modes else pm)
+
+    # Change meters to miles
+    # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
+    if "distance" in expanded_ct.columns:
```

Comment:
Similar to this block of code, we do have similar codes in the existing function load_viz_notebook_data. Could we  refactor these two functions?

Reply:
Yes, we can consider it after we get this change in. I did consider the refactoring as part of this change, but it is a bit tricky because the implementations are similar but not similar enough to easily handle both cases without a lot of if/else.

Given that this is a different data representation, I felt that it would be more clear to have a separate implementation focused on this set of metrics - in the spirit of a separate Data Access Layer that encapsulates the differences.

However, I am open to reviewing a PR that attempts to merge them without resorting to complex if/then logic in the already complex `load_viz_notebook` function. I actually think that there is more scope for refactoring at the notebook/visualization level than in the lower level functions that are actually dependent on the details of the data storage.